### PR TITLE
Goal fix, trap fix, gold despawn fix

### PR DESCRIPTION
**Pull request by ScipioWright**
Fixes each of these issues.

### CODE REVIEW

File: data/archipelago/scripts/ap_utils.lua [hunk 1]
```diff
@@ -235,7 +235,12 @@ function create_our_item_entity(item, x, y)
     	return perk_spawn(x, y, item.perk, true)
   	elseif item.items ~= nil and #item.items > 0 then
     	-- our item is something else (random choice)
-    	return EntityLoad(item.items[Random(1, #item.items)], x, y)
+		local entity_id = EntityLoad(item.items[Random(1, #item.items)], x, y)
+		if item.gold_amount ~= 0 then
+			local life_comp = EntityGetFirstComponent(entity_id, "LifetimeComponent", "enabled_in_world")
+			EntityRemoveComponent(entity_id, life_comp)
+		end
```
Comment (heinermann) [inline data/archipelago/scripts/ap_utils.lua:242]:
```suggestion
		local life_comp = EntityGetFirstComponent(entity_id, "LifetimeComponent", "enabled_in_world")
		if life_comp ~= nil then
			EntityRemoveComponent(entity_id, life_comp)
		end
```

May be a more general way to write it.

File: data/archipelago/scripts/extend_temple_altar.lua [hunk 2]
```diff
@@ -100,7 +100,7 @@ local function ap_extend_temple_altar()
 		-- This part would otherwise be a spell refresher
 
 		local location_id = AP.FIRST_SPELL_REFRESH_LOCATION_ID + get_shop_num(y) - 1
-    local is_not_obtained = Globals.MissingLocationsSet:has_key(location_id)
+    	local is_not_obtained = Globals.MissingLocationsSet:has_key(location_id)
```
Comment (heinermann) [inline data/archipelago/scripts/extend_temple_altar.lua:103]:
```suggestion
		local is_not_obtained = Globals.MissingLocationsSet:has_key(location_id)
```

Comment (heinermann) [reply data/archipelago/scripts/extend_temple_altar.lua:103]: Gonna have to do a pass on everything, replacing 2-spaces with tabs.

File: data/archipelago/scripts/item_utils.lua [hunk 3]
```diff
@@ -81,8 +80,14 @@ function NGSpawnItems(item_list)
 				end
 			end
 			item_list[item] = nil
-			-- give the player their perks
+		
+		elseif item == 110032 then
```
Comment (heinermann) [inline data/archipelago/scripts/item_utils.lua:84]: `MAP_PERK_ID = 110032` or something in constants.lua would be good.

File: init.lua [hunk 4]
```diff
@@ -89,6 +89,7 @@ local function CheckVictoryConditionFor(flag, msg)
 		Log.Info(msg)
 		SendCmd("StatusUpdate", {status = 30})
 		GameRemoveFlagRun(flag)
+		SendCmd("Say", { text = "!release"})
```
Comment (heinermann) [inline init.lua:92]: This seems weird. My thought was that we don't need to do this? AP should be doing this automatically?

File: data/archipelago/scripts/shopitem_utils.lua [hunk 5]
```diff
@@ -33,7 +33,12 @@ function ShopItems.create_our_item_entity(item, x, y)
     return perk_spawn(x, y, item.perk, true)
   elseif item.items ~= nil and #item.items > 0 then
     -- our item is something else (random choice)
-    return EntityLoad(item.items[Random(1, #item.items)], x, y)
+    local entity_id = EntityLoad(item.items[Random(1, #item.items)], x, y)
```
Comment (heinermann) [inline data/archipelago/scripts/shopitem_utils.lua:36]: Hmm gonna have to look into deduplicating this function later.

File: data/archipelago/scripts/item_utils.lua (hunk 3 above)
Comment (ScipioWright) [reply data/archipelago/scripts/item_utils.lua:84]: Will do in the next one, my pycharm is having issues. I've written up the change already (for this and the heart ID)

**Pull request merged**